### PR TITLE
Commen main

### PR DESCRIPTION
**Pull request by stped18**
I have created core.xml that dynamic bind and unbind functions in a new class ServicesStorage.
Services storage have the responsibility to add and remove services from the game.
it also has an update function that is called in the render function from the game class. to update all processes

### CODE REVIEW

File: Common/src/main/java/dk/sdu/se4/common/entity/parts/LifePart.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+public class LifePart implements EntityPart {
+
+    private int hp;
+    private int MAX_HP;
+    private boolean alive;
+
+    public LifePart(int MAX_HP) {
+        this.hp = MAX_HP;
+        this.MAX_HP = MAX_HP;
+        this.alive = true;
```
Comment (Nicklason) [inline Common/src/main/java/dk/sdu/se4/common/entity/parts/LifePart.java:23]: I think it should just be alive if the hp is above 0

File: Common/src/main/java/dk/sdu/se4/common/entity/parts/MovingPart.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    public MovingPart(float speed) {
+        this.speed = speed;
+
+    }
+
+    public Point getDeltaPoint() {
+        return deltaPoint;
+    }
+
+    public void setDeltaPoint(Point deltaPoint) {
```
Comment (Nicklason) [inline Common/src/main/java/dk/sdu/se4/common/entity/parts/MovingPart.java:30]: what does delta point mean?

File: Common/src/main/java/dk/sdu/se4/common/service/AIControlSystem.java [hunk 3]
```diff
@@ -10,5 +10,8 @@
  * @author steff
  */
 public interface AIControlSystem {
-    void pathFinding();
+
+    void addPathFinding();
+
+    void removePathFinding();
 }
```
Comment (Nicklason) [inline Common/src/main/java/dk/sdu/se4/common/service/AIControlSystem.java:17]: What does the add and remove methods do?

File: Core/src/main/java/dk/sdu/se4/core/Game.java [hunk 4]
```diff
@@ -5,25 +5,35 @@
 import com.badlogic.gdx.graphics.GL20;
 
 public class Game implements ApplicationListener {
+
+    private ServicesStorage servicesStorage;
+
     @Override
-    public void create() {}
+    public void create() {
+        this.servicesStorage = new ServicesStorage();
```
Comment (Nicklason) [inline Core/src/main/java/dk/sdu/se4/core/Game.java:13]: I still don't think you should have a class just for the services. I think it adds unnessesary complexity.

File: Core/src/test/java/dk/sdu/se4/core/TestPostProcessorService.java [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+import dk.sdu.se4.common.service.PostProcessorService;
+
+/**
+ *
+ * @author steff
+ */
+public class TestPostProcessorService implements PostProcessorService {
+
+    public void process() {
+        System.out.println("PostProcessorService is called");
```
Comment (Nicklason) [inline Core/src/test/java/dk/sdu/se4/core/TestPostProcessorService.java:17]: Should it be printing?

File: Core/src/test/java/dk/sdu/se4/core/TestProcessorService.java [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+import dk.sdu.se4.common.service.ProcessorService;
+
+/**
+ *
+ * @author steff
+ */
+public class TestProcessorService implements ProcessorService {
+
+    public void process() {
+        System.out.println("TestProcessor is called");
```
Comment (Nicklason) [inline Core/src/test/java/dk/sdu/se4/core/TestProcessorService.java:17]: Should it be printing?

**Review by Nicklason — CHANGES_REQUESTED** [review]

File: Common/src/main/java/dk/sdu/se4/common/entity/parts/LifePart.java (hunk 1 above)
Comment (stped18) [reply Common/src/main/java/dk/sdu/se4/common/entity/parts/LifePart.java:23]: do you want a check if hp >0 then alive :)?
who will create a dead Entity XD
I create a check :)

Comment (stped18) [reply Common/src/main/java/dk/sdu/se4/common/entity/parts/LifePart.java]: Done:D

File: Common/src/main/java/dk/sdu/se4/common/entity/parts/MovingPart.java (hunk 2 above)
Comment (stped18) [reply Common/src/main/java/dk/sdu/se4/common/entity/parts/MovingPart.java:30]: it the point from the whare to calculate the point

File: Common/src/main/java/dk/sdu/se4/common/service/AIControlSystem.java (hunk 3 above)
Comment (stped18) [reply Common/src/main/java/dk/sdu/se4/common/service/AIControlSystem.java:17]: adding the pathfinding and removing the partfinding module

File: Core/src/main/java/dk/sdu/se4/core/Game.java (hunk 4 above)
Comment (stped18) [reply Core/src/main/java/dk/sdu/se4/core/Game.java:13]: I think is a good choice to give the responsibility to a specified class :) ane class shot only have small responsibility

File: Core/src/test/java/dk/sdu/se4/core/TestProcessorService.java (hunk 6 above)
Comment (stped18) [reply Core/src/test/java/dk/sdu/se4/core/TestProcessorService.java:17]: I cant find another way to check if is called

Comment (stped18) [reply Core/src/test/java/dk/sdu/se4/core/TestProcessorService.java:17]: it doesn't return enything

File: Common/src/main/java/dk/sdu/se4/common/entity/parts/MovingPart.java (hunk 2 above)
Comment (stped18) [reply Common/src/main/java/dk/sdu/se4/common/entity/parts/MovingPart.java:30]: I don't use point. locate I use translate 
I will fix that

File: Core/src/test/java/dk/sdu/se4/core/TestPostProcessorService.java (hunk 5 above)
Comment (stped18) [reply Core/src/test/java/dk/sdu/se4/core/TestPostProcessorService.java:17]: only way to validatee

Comment (Nicklason) [reply Core/src/test/java/dk/sdu/se4/core/TestPostProcessorService.java:17]: couldn't you just use assertTrue(true) ?

**Pull request closed without merging**